blockchain/filter.go

```diff
@@ -0,0 +1,209 @@
+package blockchain
```

Comment:
a general note, I took the custom of adding the GLPv3 header on every "new" file (also as a marker of what was edit during the great-reordering)